File: misc/clqz/3.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(N):
    # write your code in Python 3.6
    if N<2:
        return N
    first = 0
    second = 1
    fib = 0
    count = 1
    while(count<N+1):
        fib = (first+second)%10000000
        second = first
        first = fib
        count += 1
    return fib%1000000

File: misc/frht/3.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int) int {
	// write your code in Go 1.4
	if len(A) == 0 {
		return -1
	}
	shouldBeAllEven := (A[0]%2 == 0)
	min := 200000000
	max := -200000000
	for _, val := range A {
		if (val%2 == 0) != shouldBeAllEven {
			return -1
		}
		if val > max {
			max = val
		}
		if val < min {
			min = val
		}
	}
	return (max - min) / 2
}
